flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -2474,16 +2532,19 @@ public void jobManagerLostLeadership(final JobID jobId, final JobMasterId jobMas
                     "JobManager for job {} with leader id {} lost leadership.", jobId, jobMasterId);
 
             runAsync(
-                    () ->
-                            jobTable.getConnection(jobId)
-                                    .ifPresent(
-                                            jobManagerConnection ->
-                                                    disconnectJobManagerConnection(
-                                                            jobManagerConnection,
-                                                            new Exception(
-                                                                    "Job leader for job id "
-                                                                            + jobId
-                                                                            + " lost leadership."))));
```

Comment:
Why not just passing the param `shouldRetainPartitionsOnJobManagerConnectionLost()` in?